grails-app/taglib/org/pih/warehouse/ProductTagLib.groovy

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2012 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse
+
+class ProductTagLib {
+
+    def container = { attrs, body -> }
+
+    def renderHandlingIcons = { attrs, body ->
+        if (!attrs.product) {
+            out << ""
+        } else {
+            def iconElements = ""
+            def handlingIcons = attrs.product.getHandlingIcons()
+            handlingIcons.each { it ->
```

Comment:
This is fine. The other I was thinking about doing this would have been to put all of the HTML into a GSP template file
```
out << g.render(template: '/taglib/handlingIcons', model: [attrs: attrs])
```
And then in /taglib/_handlingIcons.gsp you would just add 
```
<g:each var="handlingIcon" in="${product.handlingIcons}">
   <i class='fa ${handlingIcon.icon}' style='color: ${handlingIcon.color?:'inherit'}' ></i>&nbsp;
</g:each>
```